core/templates/pages/contributor-dashboard-page/translation-language-selector/translation-language-selector.component.spec.ts

```diff
@@ -183,6 +185,44 @@ describe('Translation language selector', () => {
     expect(component.activeLanguageCode).toBe('en');
   });
 
+  it('should display the preferred language when the preferred' +
+    ' language is defined', async(() => {
+    preferredLanguageCode = 'en';
+    let languageDescription = component.languageIdToDescription.en;
+    component.activeLanguageCode = null;
+
+    spyOn(component.setActiveLanguageCode, 'emit').and.callFake(
+      (languageCode) => {
+        component.activeLanguageCode = (
+          languageCode ? languageCode : null);
```

Comment:
Actually, I was facing problems in resolving the front-end tests so I was trying different things in order to debug the issue. I thought it would solve the issue and it did work but now I have realized that the main issue was in the `contributionOpportunitiesBackendApiServiceStub` where I have added an `if` statement (in the `getPreferredTranslationLanguageAsync` method to match with the actual function present inside `ngOnInit()`).
This ternary is irrelevant so I have removed this now.

Reply:
I added this ternary to remove the Typescript error `TS2322: Type 'string | undefined' is not assignable to type 'string | null'`. I hope it makes sense.

Reply:
You might need to declare the `languageCode` param as a string type, e.g.`callFake((languageCode: string)` to satisfy the compiler.